sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -173,6 +174,24 @@ def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
             raise ValueError("No token scope provided.")
         return _generate_sas_token(scopes[0], self.policy, self.key)
 
+class AzureNamedKeyTokenCredential(object):
+    """The named key credential used for authentication.
+
+    :param str credential: The AzureNamedKeyCredential that should be used
+    """
+
+    def __init__(self, credential):
+        # type: (AzureNamedKeyCredential) -> None
+        self.credential = credential
+        self.token_type = b"servicebus.windows.net:sastoken"
+
+    def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
+        # type: (str, Any) -> _AccessToken
+        if not scopes:
+            raise ValueError("No token scope provided.")
+        name, key = self.credential.named_key
+        return _generate_sas_token(scopes[0], name, key)
```

Comment:
nit: coding consistency with `AzureSasTokenCredential`
```suggestion
    def __init__(self, azure_named_key_credential):
        # type: (AzureNamedKeyCredential) -> None
        self._credential = azure_named_key_credential
        self.token_type = b"servicebus.windows.net:sastoken"

    def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
        # type: (str, Any) -> _AccessToken
        if not scopes:
            raise ValueError("No token scope provided.")
        name, key = self._credential.named_key
        return _generate_sas_token(scopes[0], name, key)
```